portfolio/src/main/webapp/script.js

```diff
@@ -12,31 +12,91 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+window.addEventListener('DOMContentLoaded', () => {
+  const navBar = document.getElementById('nav-bar');
+
+  Array.from(navBar.children).forEach(child => {
+      child.addEventListener('click', () => {
+          const targetId = child.getAttribute('data-target-id');
+          const targetElement = document.getElementById(targetId);
+          showContent(targetElement);
+        });
+      });
+
+  document.getElementById("comment-amount")
+      .addEventListener("change", getComments);
+    
+  document.getElementById("delete-comments")
+      .addEventListener("click", deleteComments);
+
+  getComments();
+});
+
+/** Displays the content title and text in the content container. */
+function showContent(element) {
+  const contentContainer = document.getElementById('content-container');
+
+  Array.from(contentContainer.children).forEach(child => {
+      child.classList.toggle('hidden', true);
+  });
+
+  element.classList.toggle('hidden', false);
+}
+
 /** Retrieves data from the /data page and displays it. */
 function getComments() {
-  fetch('/data').then(response => response.json()).then((json) => {
-    json.forEach(comment => displayComment(comment));
-  });
+  clearComments();
+
+  localStorage.setItem("commentAmount",
+      document.getElementById("comment-amount").value);
+
+  commentAmount = localStorage.getItem("commentAmount");
+
+  fetch(`/data?comment-amount=${commentAmount}`)
+      .then(response => response.json())
+      .then(json => {
+          json.forEach(displayComment);
+        });
+
+}
+
+/** Clears the div container holding the comments. */
+function clearComments() {
+  commentContainer = document.getElementById("comments");
+    while(commentContainer.firstChild) {
+      commentContainer.removeChild(commentContainer.firstChild);
+  }
 }
 
 /** 
  * Creates paragraph elements and sets the text to comments 
- * pulled from \data page. 
+ * pulled from /data page. 
  */
 function displayComment(comment) {
   const commentElement = document.createElement("p");
   commentElement.innerText =
-   `${comment.text} posted on: ${comment.month}/${comment.day}/${comment.year}`;
-  document.getElementById("comments-container").appendChild(commentElement);
+      `${comment.text} posted on: ${comment.month}/${comment.day}/${comment.year}`;
+  document.getElementById("comments").appendChild(commentElement);
 }
 
-window.addEventListener('DOMContentLoaded', getComments, false);
+/** Removes the comments from the page after being cleared from the Datasore. */
+function deleteComments() {
+  fetch("/data", {method: "DELETE"}).then(clearComments);
+}
+
+function loadCanvas() {
```

Comment:
Nit: This method is not used and can be removed.

Reply:
I created it but I am going to use it later.